pkg/restapi/v1/oidc4ci/openapi.gen.go

```diff
@@ -35,6 +35,66 @@ type AccessTokenResponse struct {
 	TokenType string `json:"token_type"`
 }
 
+// Client registration request.
+type ClientRegistrationRequest struct {
+	// Human-readable string name of the client to be presented to the end-user during authorization.
+	ClientName *string `json:"client_name,omitempty"`
+
+	// URL string of a web page providing information about the client.
+	ClientUri *string `json:"client_uri,omitempty"`
+
+	// Array of OAuth 2.0 grant types that the client is allowed to use. Supported values: authorization_code, urn:ietf:params:oauth:grant-type:pre-authorized_code.
+	GrantTypes []string `json:"grant_types"`
+
+	// Array of allowed redirection URI strings for the client. Required if client supports authorization_code grant type.
+	RedirectUris *[]string `json:"redirect_uris,omitempty"`
+
+	// Array of OAuth 2.0 response types that the client can use at the authorization endpoint. Supported values: code.
+	ResponseTypes *[]string `json:"response_types,omitempty"`
+
+	// String containing a space-separated list of scope values that the client can use when requesting access tokens.
+	Scope *string `json:"scope,omitempty"`
+
+	// Requested client authentication method for the token endpoint. Supported values: none, client_secret_post, client_secret_basic. None is used for public clients (native apps, mobile apps) which can not have secrets. Default: client_secret_basic.
+	TokenEndpointAuthMethod *string `json:"token_endpoint_auth_method,omitempty"`
+}
+
+// Response with registered metadata for the client.
+type ClientRegistrationResponse struct {
+	// Client identifier.
+	ClientId string `json:"client_id"`
```

Comment:
I noticed that a lot of these generated names also don't do abbreviations correctly (e.g. it shoes `Id` instead of `ID`). Is this a limitation of the automatic generation?

Reply:
[oapi-codegen](https://github.com/deepmap/oapi-codegen/) that is used in vcs allows customizing generated names with `x-go-name` custom tag, but currently we rely on default behavior. We might reconsider that in future :-)

As for types - tool uses pointers to differentiate absent of value from empty one - this is crucial for validation.

Reply:
OK, yeah I suppose customizing generate names would be a bit better, but certainly not a big deal. For the types, I understanding having `*string`, but I also see `*[]string` for `redirectUris`, which is unnecessary as slices/arrays are already nillable.

Reply:
(but, again, not really important, just something to look out for)